vnext/template/cs-app/src/App.xaml.cs

```diff
@@ -43,9 +43,23 @@ public App()
         protected override void OnLaunched(LaunchActivatedEventArgs e)
         {
             base.OnLaunched(e);
-            var frame = Window.Current.Content as Frame;
-            frame.Navigate(typeof(MainPage));
-            Window.Current.Activate();
+            var frame = (Frame)Window.Current.Content;
+            frame.Navigate(typeof(MainPage), e.Arguments);
+        }
+
+        /// <summary>
+        /// Invoked when the application is activated by some means other than normal launching.
+        /// </summary>
+        protected override void OnActivated(IActivatedEventArgs e)
+        {
+            var preActivationContent = Window.Current.Content;
+            base.OnActivated(e);
+            if (preActivationContent == null && Window.Current != null)
+            {
+                // Display the initial content
+                var frame = (Frame)Window.Current.Content;
+                frame.Navigate(typeof(MainPage), null);
```

Comment:
(nit) How would folks know to add this code in some new project? Just copy/paste from current one? This activation is completely non obvious :(

Reply:
this code is the scaffolding that gets created for them when they create a new app

Reply:
Unfortuantely we still have to ask users to manually update...

@NickGerleman: Maybe a crazy idea for a tool:
Pick the current projects file, create a template using the existing RNW version, create a template using new RNW version. Generate a patch between the two and apply to their repo?

Reply:
Not that crazy. There's some existing rn tooling that we've looked at integrating with that follows the strategy. Apparently community three way merge tooling is pretty busted, but we might be able to do a better job.

"react-native-upgrade-helper" is the current community solutions. It's just a webpage with diffs between new apps across versions. Might be hard for RNW since we have a wider matrix of flavors.